addons/knowledge/models/knowledge_article.py

```diff
@@ -830,7 +834,7 @@ def _resequence(self):
     def _get_max_sequence_inside_parents(self, parent_ids):
         max_sequence_by_parent = {}
         rg_results = self.env['knowledge.article'].sudo().read_group(
-            [('parent_id', 'in', parent_ids)],
```

Comment:
would extract the domain to build it with a clean if parent_ids: domain = gnagnagna else: domain = gnugnugnu.
A bit more verbose but cleaner I think.